src/pybind/cephfs.py

```diff
@@ -82,6 +83,12 @@ class cephfs_statvfs(Structure):
                 ("f_flag", c_uint),
                 ("f_namemax", c_uint)]
 
+class cephfs_dirent(Structure):
+    _fields_ = [("d_ino", c_long),
+                ("d_off", c_ulong),
+                ("d_reclen", c_ushort),
+                ("d_type", c_uint8),
+                ("d_name", c_char*256)]
```

Comment:
Ah, I've just looked at Client::fill_dirent, and I see that this code is also assuming that d_name is a 255 byte array.  So strictly the Client C++ part of this is not being portable.  However, since it's always our code generating these structures, we should always be able to guarantee that we pad out the names to 255 bytes with nulls, so this python part should be safe as long as we bear that in mind.